torch/ao/quantization/quantizer/x86_inductor_quantizer.py

```diff
@@ -326,7 +453,58 @@ def get_supported_operator_for_quantization_config(
                 return ops
         return []
 
+    def _get_current_quantization_mode(self) -> _QuantizationMode:
+        """Retrieves the current quantization mode based on all configurations."""
+        is_qat = None
+        is_dynamic = None
+
+        for qconfig in (
+            list(self.module_name_qconfig.values())
+            + list(self.operator_type_qconfig.values())
+            + [self.global_config]
+        ):
+            if qconfig is not None:
+                is_qat = qconfig.is_qat
+                input_activation_spec = qconfig.input_activation
+                if input_activation_spec is not None:
+                    is_dynamic = input_activation_spec.is_dynamic
```

Comment:
It looks like we only pick the last qconfig. Is it expected?

Reply:
Yes, if I understand correctly, all existing not-None `quantization_config`s should have the same `is_qat`, and all existing `quantization_config`s with not-None  `input_activation` should have the same `is_dynamic`.

By `existing configs`, I refer to the configurations within `self.module_name_qconfig`, `self.operator_type_qconfig`, or `self.global_config`.